### ISSUE TITLE
Importing and using H from woocommerce/components throws error.

### ISSUE THREAD

**Issue by leewillis77** [woocommerce/woocommerce-admin#6482]
**Describe the bug**
I'm looking into adding setup tasks as per https://developer.woocommerce.com/2021/02/26/use-setup-tasks-to-provide-a-first-class-on-boarding-experience-for-merchants/

The basics are all working fine, so now I'm trying to work on the content. In order to keep things consistent, I'm attempting to re-use as many of the WordPress/WooCommerce components as possible. For this change, I'm introducing a `CardHeader`, with enclosed `H` component into my task, e.g.

I've required the following packages in my package.json:
```
"dependencies": {
    "@woocommerce/components": "^5.1.2",
    "@woocommerce/navigation": "^5.2.0",
...
```

In my javascript I have the following:
```
import { Button, Card, CardBody, CardHeader } from '@wordpress/components';
import { H } from '@woocommerce/components';
```

My component looks something like this:
```
<Card className="woocommerce-task-card">
    <CardHeader>
        <H>
	    MyHeader
        </H>
    </CardHeader>
...
```

**Expected behavior**
Everything works fine without the `<H></H>` in the `CardHeader`, but as soon as I add the `<H>` sub-component to my card, I expect the component to render in the same style as the blocks on the main WooCommerce admin homepage, however instead my set up tasks are no longer rendered and I receive the following error in the browser console:

```
Uncaught TypeError: Cannot read property 'withViewportMatch' of undefined
    at Module../node_modules/@woocommerce/components/build-module/chart/index.js (index.js:686)
    at __webpack_require__ (bootstrap:19)
    at Module../node_modules/@woocommerce/components/build-module/index.js (index.js:1)
    at __webpack_require__ (bootstrap:19)
    at Module../js/src/setup-tasks.js (bootstrap:83)
    at __webpack_require__ (bootstrap:19)
    at bootstrap:83
    at bootstrap:83
```

**Comment by ilyasfoo** [comment]
Hi @leewillis77,

Thank you so much for filing this issue. I have tried to reproduce with a custom plugin, but I'm unable to get the browser error that you mentioned and my component rendered properly. I'm using WooCommerce `5.1.0-rc.1` and I've deactivated WooCommerce Admin plugin in my test. I'm also using the same imports that you use.

My component:
```jsx
const Task = () => {
  return (
    <Card className="woocommerce-task-card">
      <CardHeader>
        <H>MyHeader</H>
      </CardHeader>
    </Card>
  );
};

```

Render:

![image](https://user-images.githubusercontent.com/3747241/110104173-c61f8480-7de1-11eb-88be-dfdac77909ea.png)

Note that I'm building my plugin using the provided script in WooCommerce Admin: `npm run example -- --ext=add-task` on branch `release/2.0.1`. It should have the same version of `@woocommerce/components` and `@woocommerce/navigation` that you're using. Maybe you could try to duplicate the build script from our example docs?

I'm not sure how to reproduce this any other way, would be great if you could advise or maybe provide a fully built plugin. Thanks!

**Comment by leewillis77** [comment]
Hi. 

Thanks for the feedback. I've pulled together a "minimal" example of this here:
https://github.com/leewillis77/gh6482

If built with `woocommerce-admin` active, then this works, and renders an extension task on the WooCommerce home screen:
![Screenshot 2021-03-05 at 11 49 47](https://user-images.githubusercontent.com/1097338/110111933-fb5bb080-7da8-11eb-88c2-e1585285e4ad.png)

If built without `woocommerce-admin` active then this fails, and the following error is thrown in the browser console:
```
index.js:686 Uncaught TypeError: Cannot read property 'withViewportMatch' of undefined
    at Module../node_modules/@woocommerce/components/build-module/chart/index.js (index.js:686)
    at __webpack_require__ (bootstrap:19)
    at Module../node_modules/@woocommerce/components/build-module/index.js (index.js:1)
    at __webpack_require__ (bootstrap:19)
    at Module../js/src/setup-tasks.js (bootstrap:83)
    at __webpack_require__ (bootstrap:19)
    at bootstrap:83
    at bootstrap:83
```

In the course of trimming out my actual code to produce the minimal example, I think I've narrowed down the issue, and it relates to dependency inclusion. 

The [example code](https://github.com/woocommerce/woocommerce-admin/blob/main/docs/examples/extensions/add-task/woocommerce-admin-add-task-example.php#L23-L34) linked in the [original article](https://developer.woocommerce.com/2021/02/26/use-setup-tasks-to-provide-a-first-class-on-boarding-experience-for-merchants/) doesn't pull in the dependencies from the generated asset file, but hardcodes a list of dependencies. 

For some reason this works when the code is compiled with `woocommerce-admin`, but fails when compiled against just `woocommerce`. The obvious "fix" for my code is to pull in the generated list of dependencies which then works in both cases - see https://github.com/leewillis77/gh6482/blob/main/gh6482.php#L42

However, that does imply that *something* different is happening with dependency registration between the standalone woocommerce-admin and the versions included in woocommerce. 

As an aside the original example code should probably be updated to pull in the asset file dynamically. Happy to raise a PR for that if that helps?

**Comment by psealock** [comment]
>The obvious "fix" for my code is to pull in the generated list of dependencies which then works in both cases

Hi @leewillis77, since the example code post was written there is now a tool to dynamically declare Woo and WP dependencies, [Dependency Extraction Webpack Plugin](https://github.com/woocommerce/woocommerce-admin/tree/main/packages/dependency-extraction-webpack-plugin)

Also useful is `npm run create-wc-extension` to scaffold out a new project with the dependency tool already added. Read more on that [here](https://github.com/woocommerce/woocommerce-admin/tree/main/bin/starter-pack).

I used the script to see if I can reproduce this issue. I added this code and was able to see the resulting console without error using WC-Admin activated and deactivated.

```js
/**
 * External dependencies
 */
import { Card, CardHeader } from '@wordpress/components';
import { H } from '@woocommerce/components';

const Task = () => {
	return (
		<Card className="woocommerce-task-card">
			<CardHeader>
				<H>MyHeader</H>
			</CardHeader>
		</Card>
	);
};

console.log( Task() );
```

Looking at `build/index.asset.php` to see the generated dependencies I see this:

```
<?php return array('dependencies' => array('wc-components', 'wp-components', 'wp-element'), 'version' => '9ca69248d8333f4c66b0f62ce312a3e1');
```

It looks like `wp-element` is missing from your list of dependencies. Can you try adding if that is the issue? If not, I can push up my code so you have an example. In any case, I suggest using the Depedency plugin because these can get tricky with loading order and there may be other issues also at play.

**Comment by leewillis77** [comment]
Hi; 

Thanks for that. I'm already aware of that, and indeed in my proof-of-concept code I noted that [pulling in the dependencies from the generated asset file](https://github.com/leewillis77/gh6482/blob/main/gh6482.php#L53) works both with / without WooCommerce Admin activated. 

However, I thought that there was still an issue though as the (smaller) static list of assets worked with WooCommerce Admin activated, but not without which implied a difference between WooCommerce Admin run-as-a-plugin vs bundled with WooCommerce? However, I suspect it's just a version difference (WooCommerce included 2.0.1, but I was using `main` branch of WooCommerce Admin).

I'll close this (as the original issue is resolved). As an aside though the original example code should probably be updated to pull in the asset file dynamically. Happy to raise a PR for that if that helps?

**Comment by psealock** [comment]
>Happy to raise a PR for that if that helps?

Awesome, that would be fantastic!
